homeassistant/components/websocket_api/http.py

```diff
@@ -328,7 +328,13 @@ async def async_handle(self) -> web.WebSocketResponse:
         if TYPE_CHECKING:
             assert writer is not None
 
-        send_bytes_text = partial(writer.send, binary=False)
+        # aiohttp 3.11.0 changed the method name from _send_frame to send_frame
+        if hasattr(writer, "send_frame"):
+            send_frame = writer.send_frame  # pragma: no cover
+        else:
+            send_frame = writer._send_frame  # noqa: SLF001
```

Comment:
If the fallback is also optimized (ie not using old send), it should be mentioned in the pull.

Ps . Isnt removal of send() here https://github.com/aio-libs/aiohttp/pull/9348/files#diff-37008174f47abcf385c9bd15d713b586a921dc6aad7d75acd1510aebc75d2b4fR407 unnecessary breaking change?

Reply:
Send is not a public api.  It’s accessed via a protected method

Reply:
> If the fallback is also optimized (ie not using old send), it should be mentioned in the pull.
> 
There shouldn’t be any significant performance difference.

> Ps . Isnt removal of send() here https://github.com/aio-libs/aiohttp/pull/9348/files#diff-37008174f47abcf385c9bd15d713b586a921dc6aad7d75acd1510aebc75d2b4fR407 unnecessary breaking change?

`send` is not a public api so hopefully we are the only ones calling it directly.  It’s accessed via a protected method. We are only using it today because we had no alternative.